trio_websocket/__init__.py

```diff
@@ -153,6 +156,90 @@ def _url_to_host(url, ssl_context):
     return url.host, url.port, resource, ssl_context
 
 
+def wrap_client_stream(nursery, stream, host, resource):
+    '''
+    Wrap an arbitrary stream in a client-side ``WebSocketConnection``.
+
+    This is a low-level function only needed in rare cases. Most users should
+    call ``open_websocket()`` or ``open_websocket_url()``.
+
+    The returned connection object is not immediately usable: the caller should
+    wait for the ``open_handshake`` event before trying to use the connection.
```

Comment:
The caller just needs to know when it's done. The `start()` protocol makes more sense here, but I ran into a weird case in the unit test: I want to start a server, then start a client, and have them handshake. But if `wrap_server_stream()` doesn't resume until the opening handshake completes, then it will deadlock because the client never gets a chance to start... Maybe I should redesign my test case...